packages/client/src/components/app/CodeGenerator.svelte

```diff
@@ -0,0 +1,146 @@
+<script lang="ts">
+  import { getContext } from "svelte"
+  import { onMount, afterUpdate } from "svelte"
+  import JsBarcode from "jsbarcode"
+  import { createQrSvgString } from "@svelte-put/qr"
+  import "@spectrum-css/vars/dist/spectrum-global.css"
+
+  export let value: string
+  export let codeType: "QR Code" | "Barcode"
+  export let showValue: boolean
+  export let showLogo: boolean
+  export let customLogo: string
+  export let size: number
+  export let primColor: string
+  export let vertical: boolean
+
+  const { styleable } = getContext("sdk")
+  const component = getContext("component")
+
+  const generateBarcode = () => {
+    let barcodeSize = size / 100
+    if (value) {
+      JsBarcode("#barcode-img", value, {
+        displayValue: false, // Hide the library's built in value, optionally display it later
+        width: barcodeSize,
+        height: size,
+      })
+    }
+  }
+
+  let qrContainer: HTMLElement
+
+  const generateQr = () => {
+    if (qrContainer && codeType === "QR Code" && value) {
+      const svg = createQrSvgString({
+        data: value,
+        logo: showLogo ? customLogo : "",
+        moduleFill: primColor,
+        anchorOuterFill: primColor,
+        anchorInnerFill: primColor,
+        width: size,
+        height: size,
+      })
+      qrContainer.innerHTML = svg
+    }
+  }
+
+  onMount(() => {
+    if (codeType === "Barcode") {
+      generateBarcode()
+    } else {
+      generateQr()
+    }
+  })
+
+  afterUpdate(() => {
+    if (codeType === "Barcode") {
+      generateBarcode()
+    } else {
+      generateQr()
+    }
+  })
+</script>
+
+<div
+  class="overall {vertical ? 'vertical' : ''}"
+  use:styleable={$component.styles}
+  styles="border: 3px solid red; width: 100px; height: 200px;"
+>
+  {#if value}
+    {#if codeType === "QR Code"}
+      <div class="qr-container">
+        <div bind:this={qrContainer} />
+        <div class="qr-value" style="color: {primColor}; max-width: {size}px;">
+          {showValue ? value : ""}
+        </div>
+      </div>
+    {:else}
+      <div id="barcode-container">
+        <div id="logo-and-barcode" style="height: {size}px, width: 100%">
```

Comment:
I can see that you're using quite a lot of ids here. You should never use the same ID more than once on a page, but adding multiple of this component will create multiple elements with the same ID.

The 2 options here are:
- Generate a unique ID for each instance of the component. This is harder to style because you'll need to get this ID into your styles section, so I don't recomment this approach.
- Don't use ID attributes at all. You can style elements using classes, and you can get the reference to your DOM element for jsbarcode by using binding, like you do with the qrContainer. I checked the source and it looks like it supports passing a DOM element in as the parameter instead of a string selector.